aws/table_aws_vpc_verified_access_trust_provider.go

```diff
@@ -0,0 +1,186 @@
+				Description: "The identifier to be used when working with policy rules.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "trust_provider_type",
+				Description: "The type of Verified Access trust provider.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "user_trust_provider_type",
+				Description: "The type of user-based trust provider.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "oidc_options",
+				Description: "The OpenID Connect details for an oidc-type, user-identity based trust provider.",
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "tags_src",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("Tags"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: resourceInterfaceDescription("title"),
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromP(trustProviderTurbotData, "Title"),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromP(trustProviderTurbotData, "Tags"),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listVpcVerifiedAccessTrustProviders(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	// Create session
+	svc, err := EC2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_vpc_verified_access_trust_provider.listVpcVerifiedAccessTrustProviders", "connection_error", err)
+		return nil, err
+	}
+
+	// Limiting the results
+	maxLimit := int32(200)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 5 {
+				maxLimit = 5
+			} else {
+				maxLimit = limit
+			}
+		}
+	}
+
+	input := &ec2.DescribeVerifiedAccessTrustProvidersInput{
+		MaxResults: aws.Int32(maxLimit),
+	}
+
+	if d.KeyColumnQualString("verified_access_trust_provider_id") != "" {
+		input.VerifiedAccessTrustProviderIds = []string{d.KeyColumnQualString("verified_access_trust_provider_id")}
+	}
+
+	for {
+		// List call
+		resp, err := svc.DescribeVerifiedAccessTrustProviders(ctx, input)
+
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_vpc_verified_access_trust_provider.listVpcVerifiedAccessTrustProviders", "api_error", err)
+			return nil, nil
+		}
+
+		for _, provider := range resp.VerifiedAccessTrustProviders {
+			d.StreamListItem(ctx, provider)
+
+			// Context may get cancelled due to manual cancellation or if the limit has been reached
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+		if resp.NextToken == nil {
+			break
+		} else {
+			input.NextToken = resp.NextToken
+		}
+	}
+
+	return nil, err
+}
+
+//// TRANSFORM FUNCTIONS
+
+func trustProviderTurbotData(_ context.Context, d *transform.TransformData) (interface{}, error) {
```

Comment:
Why do we need to return title from here since it looks like we're just setting it to 1 property?

Also, if we do need a separate function for getting the title, we should have 2 separate functions for the title and tags. Combining into 1 function makes the function's purpose ambiguous.

Reply:
Added two separate the transform function for tags and title.